src/Storages/StorageURL.cpp

```diff
@@ -185,11 +186,13 @@ namespace
                         std::string user_info = request_uri.getUserInfo();
                         if (!user_info.empty())
                         {
-                            std::size_t n = user_info.find(':');
+                            std::string url_decoded_user_info;
+                            Poco::URI::decode(user_info, url_decoded_user_info);
+                            std::size_t n = url_decoded_user_info.find(':');
                             if (n != std::string::npos)
                             {
-                                credentials.setUsername(user_info.substr(0, n));
-                                credentials.setPassword(user_info.substr(n + 1));
+                                credentials.setUsername(url_decoded_user_info.substr(0, n));
+                                credentials.setPassword(url_decoded_user_info.substr(n + 1));
                             }
```

Comment:
Correct variant is 
```cpp
                        if (!user_info.empty())
                        {
                            std::string::size_t pos = user_info.find(':');

                            if (pos != std::string::npos)
                            {
                                std::string decoded_user;
                                std::string decoded_password;
                                Poco::URI::decode(user_info.substr(0, n), decoded_user);
                                Poco::URI::decode(user_info.substr(n + 1), decoded_password);
                                credentials.setUsername(decoded_user);
                                credentials.setPassword(decoded_password);
                            }
                            else
                            {
                                std::string decoded_user;
                                Poco::URI::decode(user_info, decoded_user);
                                credentials.setUsername(decoded_user);
                            }
                        }
```

You do decoding first (that is wrong - imagine you have a usename with colon inside). 
And you don't have else part (you can have use w/o password).